src/Illuminate/Session/Middleware/StartSession.php

```diff
@@ -243,4 +252,16 @@ protected function usingCookieSessions()
 
         return $this->manager->driver()->getHandler() instanceof CookieSessionHandler;
     }
+
+    /**
+     * Determines if the specified session needs garbage collection for expired
+     * sessions.
+     *
+     * @param  \Illuminate\Session\SessionInterface  $session
+     * @return bool
+     */
+    protected function gcNeeded(SessionInterface $session)
+    {
+        return ! ($session->getHandler() instanceof CacheBasedSessionHandler);
```

Comment:
> Don't you have a style guide listing all those little things beyond PSR-2 ?

Yes, but we don't have fixers for this yet. StyleCI fixes 95% of our code style for us.